build/BuildSteps.cs

```diff
@@ -152,6 +155,129 @@ public static void Test()
             Shell.Run("dotnet", $"test {Settings.TestProjectFile}");
         }
 
+        public static void GenerateZipToSign()
+        {
+            foreach (var supportedRuntime in Settings.SignInfo.RuntimesToSign)
+            {
+                var targetDir = Path.Combine(Settings.OutputDir, supportedRuntime);
+                Directory.CreateDirectory(Path.Combine(targetDir, Settings.SignInfo.ToSignDir));
+
+                var toSignPaths = Settings.SignInfo.authentiCodeBinaries.Select(el => Path.Combine(targetDir, el));
+                // Grab all the files and filter the extensions not to be signed
+                var toSignFiles = FileHelpers.GetAllFilesFromFilesAndDirs(toSignPaths).Where(file => !Settings.SignInfo.filterExtenstionsSign.Any(ext => file.EndsWith(ext)));
+                FileHelpers.CreateZipFile(toSignFiles, targetDir, Path.Combine(targetDir, Settings.SignInfo.ToSignDir, Settings.SignInfo.ToSignZipName));
+
+                var toSignThirdPartyPaths = Settings.SignInfo.thirdPartyBinaries.Select(el => Path.Combine(targetDir, el));
+                // Grab all the files and filter the extensions not to be signed
+                var toSignThirdPartyFiles = FileHelpers.GetAllFilesFromFilesAndDirs(toSignThirdPartyPaths).Where(file => !Settings.SignInfo.filterExtenstionsSign.Any(ext => file.EndsWith(ext)));
+                FileHelpers.CreateZipFile(toSignThirdPartyFiles, targetDir, Path.Combine(targetDir, Settings.SignInfo.ToSignDir, Settings.SignInfo.ToSignThirdPartyName));
```

Comment:
What happens to the rest of the files. You get the files to sign and then third party binaries. But it can probably miss unknown or new binaries